assembl/static2/js/app/components/debate/brightMirror/sideComment/innerBoxView.jsx

```diff
@@ -4,71 +4,120 @@ import React, { Fragment } from 'react';
 import moment from 'moment';
 import activeHtml from 'react-active-html';
 import { I18n } from 'react-redux-i18n';
+import { Button, OverlayTrigger, Popover, Overlay } from 'react-bootstrap';
 
 import AvatarImage from '../../../common/avatarImage';
 import { transformLinksInHtml /* getUrls */ } from '../../../../utils/linkify';
 import { postBodyReplacementComponents } from '../../common/post/postBody';
+import { getConnectedUserId } from '../../../../utils/globalFunctions';
+import { editFictionCommentTooltip } from '../../../common/tooltips';
+import Permissions, { connectedUserCan } from '../../../../utils/permissions';
 
 export type Props = {
   contentLocale: string,
   extractIndex: number,
   extracts: ?Array<FictionExtractFragment>,
   comment: ExtractComment,
-  changeCurrentExtract: (?number) => void
+  changeCurrentExtract: (?number) => void,
+  setEditMode: ?(number, string) => void
 };
 
-const renderRichtext = (text: string) => activeHtml(text && transformLinksInHtml(text), postBodyReplacementComponents());
+export type State = {
+  menuTarget: HTMLElement | null
+};
+
+class InnerBoxView extends React.Component<Props, State> {
+  state = {
+    menuTarget: null
+  };
+
+  renderRichtext = (text: string) => activeHtml(text && transformLinksInHtml(text), postBodyReplacementComponents());
 
-const InnerBoxView = ({ contentLocale, extractIndex, extracts, comment, changeCurrentExtract }: Props) => {
-  const displayName =
-    comment && comment.creator && !comment.creator.isDeleted ? comment.creator.displayName : I18n.t('deletedUser');
+  render() {
+    const { contentLocale, extractIndex, extracts, comment, changeCurrentExtract, setEditMode } = this.props;
+    const { menuTarget } = this.state;
+    const displayName =
+      comment && comment.creator && !comment.creator.isDeleted ? comment.creator.displayName : I18n.t('deletedUser');
+    const canEdit =
```

Comment:
see line 53: it implies that 'comment' can not be null ==> it is not necessary to check the comment and the comment creator here